lib/private/Group/Manager.php

```diff
@@ -443,21 +443,17 @@ public function displayNamesInGroup($gid, $search = '', $limit = -1, $offset = 0
 
 		if (!empty($search)) {
 			// only user backends have the capability to do a complex search for users
-			$searchOffset = 0;
-			$searchLimit = $limit * 100;
-			if ($limit === -1) {
-				$searchLimit = 500;
-			}
-
+			$chunkOffset = 0;
+			$chunkLimit = 500;
 			do {
-				$filteredUsers = $this->userManager->searchDisplayName($search, $searchLimit, $searchOffset);
-				foreach ($filteredUsers as $filteredUser) {
-					if ($group->inGroup($filteredUser)) {
-						$groupUsers[] = $filteredUser;
+				$userChunk = $group->searchUsers('', $chunkLimit, $chunkOffset);
```

Comment:
As I understand it, `searchUsers` allows to filter by uid and display name. 
I guess we could filter a bit with it:
```suggestion
				$userChunk = $group->searchUsers($search, $chunkLimit, $chunkOffset);
```